store/storetest/job_store.go

```diff
@@ -261,6 +262,54 @@ func testJobStoreGetNewestJobByStatusAndType(t *testing.T, ss store.Store) {
 	assert.Nil(t, received)
 }
 
+func testJobStoreGetNewestJobByStatusesAndType(t *testing.T, ss store.Store) {
+	jobType1 := model.NewId()
+	jobType2 := model.NewId()
+	status1 := model.NewId()
+	status2 := model.NewId()
+
+	jobs := []*model.Job{
+		{
+			Id:       model.NewId(),
+			Type:     jobType1,
+			CreateAt: 1001,
+			Status:   status1,
+		},
+		{
+			Id:       model.NewId(),
+			Type:     jobType1,
+			CreateAt: 1000,
+			Status:   status1,
+		},
+		{
+			Id:       model.NewId(),
+			Type:     jobType2,
+			CreateAt: 1003,
+			Status:   status1,
+		},
+		{
+			Id:       model.NewId(),
+			Type:     jobType1,
+			CreateAt: 1004,
+			Status:   status2,
+		},
+	}
+
+	for _, job := range jobs {
+		_, err := ss.Job().Save(job)
+		require.Nil(t, err)
+		defer ss.Job().Delete(job.Id)
+	}
+
+	received, err := ss.Job().GetNewestJobByStatusesAndType([]string{status1, status2}, jobType1)
+	assert.Nil(t, err)
+	assert.EqualValues(t, jobs[3].Id, received.Id)
+
```

Comment:
Not required but I don't think it would hurt to add a couple more test like the following

```
received, err := ss.Job().GetNewestJobByStatusesAndType([]string{status2}, jobType2)
// No error and received job should be Nil

received, err := ss.Job().GetNewestJobByStatusesAndType([]string{status1}, jobType2)
// No error and received back jobs[2].id

received, err := ss.Job().GetNewestJobByStatusesAndType([]string{}, jobType1)
// No error and should have received back a nil
```
Not crucial but just for a little bit more of a reassurance.

Reply:
Added additional tests.